File: Objective-C/QMUIKit_Study/QMUIKit_Study/ViewController.m
//
//  ViewController.m
//  QMUIKit_Study
//
//  Created by zidonj on 2019/11/24.
//  Copyright © 2019 zidonj. All rights reserved.
//

#import "ViewController.h"
#import <QMUIKit/QMUIKit.h>
#import "PViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
        
}

//- (void)viewWillAppear:(BOOL)animated {
//    [super viewWillAppear:animated];
//    [self.navigationController setNavigationBarHidden:YES animated:animated];
//}

- (IBAction)test:(id)sender {
    PViewController *pv = PViewController.new;
    pv.hidesBottomBarWhenPushed = YES;
    [self.navigationController pushViewController:pv animated:YES];
}

#pragma mark - QMUINavigationControllerDelegate

- (UIImage *)navigationBarBackgroundImage {
    return [UIImage qmui_imageWithColor:[UIColor greenColor]];
}

- (nullable UIColor *)navigationBarBarTintColor {
    return [UIColor greenColor];
}

- (UIImage *)navigationBarShadowImage {
    return [UIImage qmui_imageWithColor:[UIColor blueColor]];;
}

- (UIColor *)navigationBarTintColor {
    return NavBarTintColor;
}

- (UIColor *)titleViewTintColor {
    return NavBarTitleColor;
}

#pragma mark - <QMUICustomNavigationBarTransitionDelegate>

- (NSString *)customNavigationBarTransitionKey {
    return @"123";
}


@end

File: Objective-C/Bolts_study/Bolts_study/ViewController.swift
//
//  ViewController.swift
//  Bolts_study
//
//  Created by zidonj on 2020/6/3.
//  Copyright © 2020 zidonj. All rights reserved.
//

import UIKit
import Bolts

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.getNumberAsync().continueWith { (task: BFTask!) -> Any? in
            print(":::::",task.result!)
        }
    }
    
    func getStringAsync() -> BFTask<AnyObject> {
        //Let's suppose getNumberAsync returns a BFTask whose result is an NSNumber.
        return self.getNumberAsync().continueWith {
            (task: BFTask!) -> NSString in
            
            let number = task.result
            return NSString(format:"%@", number as! NSNumber)
        }
    }
    
    func getNumberAsync() -> BFTask<AnyObject> {
        let bft = BFTaskCompletionSource<AnyObject>()
        bft.set(result: NSNumber(value: 8.8))
        return bft.task
    }
}

